SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -21,62 +22,95 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("No need to queue newly authoritative entity %lld because this worker is already authoritative."), AuthOp.entity_id);
-			return;
-		}
-		QueueAclAssignmentRequest(AuthOp.entity_id);
+void SpatialLoadBalanceEnforcer::OnAclAuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
+{
+	// This class should only be informed of ACL authority changes.
+	check(AuthOp.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+
+	if (AclAssignmentRequestIsQueued(AuthOp.entity_id) && AuthOp.authority != WORKER_AUTHORITY_AUTHORITATIVE)
+	{
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Log,
+			TEXT("ACL authority lost for entity %lld. Can no longer enforce the previous request for this entity."),
+			AuthOp.entity_id);
+		AclWriteAuthAssignmentRequests.Remove(AuthOp.entity_id);
+		return;
 	}
+
+	MaybeQueueAclAssignmentRequest(AuthOp.entity_id);
 }
 
-// QueueAclAssignmentRequest is called from three places.
+// MaybeQueueAclAssignmentRequest is called from three places.
 // 1) AuthorityIntent change - Intent is not authoritative on this worker - ACL is authoritative on this worker.
 //    (another worker changed the intent, but this worker is responsible for the ACL, so update it.)
 // 2) ACL change - Intent may be anything - ACL just became authoritative on this worker.
 //    (this worker just became responsible, so check to make sure intent and ACL agree.)
 // 3) AuthorityIntent change - Intent is authoritative on this worker but no longer assigned to this worker - ACL is authoritative on this worker.
 //    (this worker had responsibility for both and is giving up authority.)
-void SpatialLoadBalanceEnforcer::QueueAclAssignmentRequest(const Worker_EntityId EntityId)
+// Queuing an ACL assignment request may not occur if the assignment is the same as before, or if the request is already queued,
+// or if we don't meet the predicate required to enforce the assignment.
+void SpatialLoadBalanceEnforcer::MaybeQueueAclAssignmentRequest(const Worker_EntityId EntityId)
 {
-	// TODO(zoning): measure the performance impact of this.
-	if (AclWriteAuthAssignmentRequests.ContainsByPredicate([EntityId](const WriteAuthAssignmentRequest& Request) { return Request.EntityId == EntityId; }))
+	if (!CanEnforce(EntityId))
 	{
-		UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("An ACL assignment request already exists for entity %lld on worker %s."), EntityId, *WorkerId);
+		return;
 	}
-	else
+
+	const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(EntityId);
+	const PhysicalWorkerName* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
+	if (OwningWorkerId != nullptr &&
+		*OwningWorkerId == WorkerId &&
+		StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
```

Comment:
Looks like indenting is off here?

Reply:
it's indented a tab length in for each new line inside the if. Changed to put the && at the start of each new line